pymeasure/instruments/oxfordinstruments/mercuryitc.py

```diff
@@ -0,0 +1,314 @@
+
+    Depending on sensor type (e.g. diode, NTC/PTC resistor), some properties
+    will not be valid. Therefore, sensor configuration must be known and set
+    a priori. Only diode sensor types are currently implemented.
+
+    Control loops are accessed via temeperature sensors, rather than heaters.
+    Therefore loop control is kept with other temperature sensor properties.
+    """
+
+    def check_set_errors(self):
+        """The MercuryiTC responds to every communication (get and set).
+        This simply reads back on setting a property."""
+        try:
+            self.read()
+        except Exception as exc:
+            log.exception("Setting a property failed.", exc_info=exc)
+            raise
+        else:
+            return []
+
+    temperature = Channel.measurement(
+        "READ:DEV:{ch}:TEMP:SIG:TEMP",
+        """Get the measured temperature, in Kelvin.""",
+        preprocess_reply=lambda v: v.split(":")[-1].replace("K", "")
+    )
+
+    voltage = Channel.measurement(
+        "READ:DEV:{ch}:TEMP:SIG:VOLT",
+        """Get the sensor voltage, in Volts.""",
+        preprocess_reply=lambda v: v.split(":")[-1].replace("mV", ""),
+        get_process=lambda v: v*1e-3
+    )
+
+    control_temperature = Channel.measurement(
+        "READ:DEV:{ch}:TEMP:SIG:TEMP",
+        """Get the control temperature, in Kelvin.""",
+        preprocess_reply=lambda v: v.split(":")[-1].replace("K", "")
+    )
+
+    loop_PID_enabled = Channel.control(
+        "READ:DEV:{ch}:TEMP:LOOP:ENAB",
+        "SET:DEV:{ch}:TEMP:LOOP:ENAB:%s",
+        """Control whether the control-loop heater is controlled by
+        the PID loop (``True``) or the manual heater percentage (``False``).""",
+        check_set_errors=True,
+        preprocess_reply=lambda v: v.split(":")[-1],
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    loop_P = Channel.control(
+        "READ:DEV:{ch}:TEMP:LOOP:P",
+        "SET:DEV:{ch}:TEMP:LOOP:P:%g",
+        """Control the proportional term, P, of the control loop.""",
+        check_set_errors=True,
+        preprocess_reply=lambda v: v.split(":")[-1]
+    )
+
+    loop_I = Channel.control(
+        "READ:DEV:{ch}:TEMP:LOOP:I",
+        "SET:DEV:{ch}:TEMP:LOOP:I:%g",
+        """Control the integral term, I, of the control loop.""",
+        check_set_errors=True,
+        preprocess_reply=lambda v: v.split(":")[-1]
+    )
+
+    loop_D = Channel.control(
+        "READ:DEV:{ch}:TEMP:LOOP:D",
+        "SET:DEV:{ch}:TEMP:LOOP:D:%g",
+        """Control the derivative term, D, of the control loop.""",
+        check_set_errors=True,
+        preprocess_reply=lambda v: v.split(":")[-1]
+    )
+
+    loop_T_setpoint = Channel.control(
+        "READ:DEV:{ch}:TEMP:LOOP:TSET",
+        "SET:DEV:{ch}:TEMP:LOOP:TSET:%g",
+        """Control the control loop temperature setpoint, in Kelvin (in the range
```

Comment:
That is, how we propose the text:
```suggestion
        """Control the control loop temperature setpoint, in Kelvin. (float, truncated between 0-2000)
```

If you change the validator, replace truncated to strict